aws/resource_aws_ec2_client_vpn_network_association.go

```diff
@@ -29,14 +30,19 @@ func resourceAwsEc2ClientVpnNetworkAssociation() *schema.Resource {
 			},
 			"security_groups": {
 				Type:     schema.TypeSet,
-				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
 				Computed: true,
+				ForceNew: true,
```

Comment:
It'd be nice if we implemented the `Update` function and removed `ForceNew: true` for this attribute. 😄 `ApplySecurityGroupsToClientVpnTargetNetwork` replaces existing security groups with new ones so should be straightforward, e.g.

```go
func resourceAwsEc2ClientVpnNetworkAssociationUpdate(d *schema.ResourceData, meta interface{}) error {
  conn := meta.(*AWSClient).ec2conn

  if d.HasChange("security_groups") {
    input := &ec2.ApplySecurityGroupsToClientVpnTargetNetworkInput{
      ClientVpnEndpointId: aws.String(d.Get("client_vpn_endpoint_id").(string)),
      SecurityGroupIds:    expandStringSet(d.Get("security_groups").(*schema.Set)),
      VpcId:               aws.String(d.Get("vpc_id").(string)),
    }
    
    if _, err := conn.ApplySecurityGroupsToClientVpnTargetNetwork(input); err != nil {
      return fmt.Errorf("error applying security groups to Client VPN Target Network: %s", err)
    }
  }

  return nil
}
```

Reply:
Oops, I missed that part.  This has also been implemented with my latest commit.